// @react-login-page/base/cjs/control/Password.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Password = void 0;
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _reactLoginPage = require("react-login-page");
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["keyname", "name", "rename"];
var Password = exports.Password = function Password(props) {
  var _props$keyname = props.keyname,
    keyname = _props$keyname === void 0 ? 'password' : _props$keyname,
    name = props.name,
    rename = props.rename,
    elmProps = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var nameBase = name || rename || keyname;
  var key = keyname || name;
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactLoginPage.Input, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    type: "password",
    placeholder: "Password",
    index: 2,
    autoComplete: "on"
  }, elmProps), {}, {
    name: nameBase,
    keyname: key
  }));
};
Password.displayName = 'BaseLogin.Password';
