// @react-login-page/base/cjs/control/Username.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Username = void 0;
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _reactLoginPage = require("react-login-page");
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["keyname", "name", "rename"];
var Username = exports.Username = function Username(props) {
  var _props$keyname = props.keyname,
    keyname = _props$keyname === void 0 ? 'username' : _props$keyname,
    name = props.name,
    rename = props.rename,
    elmProps = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var nameBase = name || rename || keyname;
  var key = keyname || name;
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactLoginPage.Input, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    placeholder: "Username",
    spellCheck: false,
    index: 1
  }, elmProps), {}, {
    name: nameBase,
    keyname: key
  }));
};
Username.displayName = 'BaseLogin.Username';
